File: backend/schemas/imports.py
"""Manual import request DTOs (docs/api-project.md §3)."""

import uuid
from typing import Optional

from pydantic import AwareDatetime, BaseModel


class ImportCreateModel(BaseModel):
    original_filename: str


class ImportCompleteModel(BaseModel):
    boat_id: uuid.UUID
    activity_id: Optional[uuid.UUID] = None
    session_id: Optional[uuid.UUID] = None
    subject_type: str = "boat"  # boat | crew_member
    subject_user_id: Optional[uuid.UUID] = None
    started_at: Optional[AwareDatetime] = None  # fallback when the file has no timestamps

File: frontend/src/services/imports.ts
import { api } from "@/api/client";
import type { ImportRow, ImportTicket, UUID } from "@/types";

export const importKeys = {
  all: ["imports"] as const,
  detail: (id: UUID) => ["imports", id] as const,
};

export const importsService = {
  /** Step 1: register the import → PUT the file bytes to `upload_url`. */
  create: (original_filename: string) => api.post<ImportTicket>("/imports", { original_filename }),

  /** Step 2: bind to a boat and start processing. */
  complete: (
    id: UUID,
    body: {
      boat_id: UUID;
      activity_id?: UUID;
      session_id?: UUID;
      subject_type?: "boat" | "crew_member";
      subject_user_id?: UUID;
      started_at?: string;
    },
  ) => api.post<ImportRow>(`/imports/${id}/complete`, body),

  list: () => api.get<ImportRow[]>("/imports"),
  get: (id: UUID) => api.get<ImportRow>(`/imports/${id}`),
};
